timbuctoo-instancev4/src/main/java/nl/knaw/huygens/timbuctoo/rml/rmldata/builders/TermMapBuilder.java

```diff
@@ -90,17 +92,29 @@ public void withTemplateTerm(RdfResource column, Set<RdfResource> termTypes, Set
     withTemplateTerm(value, termType, language, datatype);
   }
 
-  public void withTemplateTerm(String value, Optional<TermType> termType, Optional<String> language,
-                             Optional<String> datatype) {
-    if (termType.isPresent()) {
-      instance = new RrTemplate(value, termType.get());
-    } else {
-      instance = new RrTemplate(value, getDefaultTermType(false, language.isPresent(), datatype.isPresent()));
-    }
+  public void withTemplateTerm(String value, Optional<TermType> termTypeOpt, Optional<String> language,
+                             Optional<String> datatypeOpt) {
+    TermType termType = termTypeOrDefault(termTypeOpt, language, datatypeOpt);
+    RDFDatatype dataType = dataTypeOrDefault(datatypeOpt);
+    instance = new RrTemplate(value, termType, dataType);
   }
 
   RrTermMap build() {
     return instance;
   }
 
+  private TermType termTypeOrDefault(Optional<TermType> termTypeOpt, Optional<String> language,
+                                     Optional<String> datatypeOpt) {
+    return termTypeOpt.orElseGet(() -> {
+      return getDefaultTermType(true, language.isPresent(), datatypeOpt.isPresent());
+    });
+  }
+
+  private RDFDatatype dataTypeOrDefault(Optional<String> datatypeOpt) {
+    if (datatypeOpt.isPresent()) {
```

Comment:
Dit moet zijn:
datatype is string, tenzij we een datype

langstrings ondersteunen we nog even niet.